AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -109,46 +108,53 @@ - (void)setParent:(id<ASLayoutable>)parent
 
 - (void)setChild:(id<ASLayoutable>)child;
 {
-  [self setChild:child forIdentifier:kDefaultChildKey];
+  ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  
+  id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
+  _child = finalLayoutable;
+  [self propagateUpLayoutable:finalLayoutable];
 }
 
 - (void)setChild:(id<ASLayoutable>)child forIdentifier:(NSString *)identifier
 {
   ASDisplayNodeAssert(self.isMutable, @"Cannot set properties when layout spec is not mutable");
+  
   id<ASLayoutable> finalLayoutable = [self layoutableToAddFromLayoutable:child];
-  self.layoutChildren[identifier] = finalLayoutable;
-  if ([finalLayoutable isKindOfClass:[ASLayoutSpec class]]) {
-    [(ASLayoutSpec *)finalLayoutable setParent:self]; // This will trigger upward propogation if needed.
-  } else if ([self supportsUpwardPropagation]) {
-    ASEnvironmentStatePropagateUp(self, finalLayoutable.environmentState.layoutOptionsState); // Probably an ASDisplayNode
-  }
+  self.childrenWithIdentifier[identifier] = finalLayoutable;
+  
+  // TODO: Should we propagate up the layoutable at it could happen that multiple children will propagated up their
+  //       layout options and one child will overwrite values from another child
+  // [self propagateUpLayoutable:finalLayoutable];
```

Comment:
@appleguy Should we propagate up the layoutable at it could happen that multiple children will propagated up their layout options and one child will overwrite values from another child?